src/WebService/Client.php

```diff
@@ -396,16 +396,31 @@ private function handleUnexpectedStatus($statusCode, $service, $path)
     }
 
     /**
-     * @param string $body    the successful request body
-     * @param string $service the service name
+     * @param int    $statusCode the HTTP status code
+     * @param string $body       the successful request body
+     * @param string $service    the service name
      *
-     * @throws WebServiceException if the request body cannot be decoded as
-     *                             JSON
+     * @throws WebServiceException if a response body is included but not
+     *                             expected, or is not expected but not
+     *                             included, or is expected and included
+     *                             but cannot be decoded as JSON
      *
      * @return array the decoded request body
      */
-    private function handleSuccess($body, $service)
+    private function handleSuccess($statusCode, $body, $service)
     {
+        $expectResponseBody = $statusCode === 204 ? false : true;
+        if (!$expectResponseBody) {
+            if (\strlen($body) !== 0) {
+                throw new WebServiceException(
+                    "Received a 204 response for $service along with an " .
+                    "unexpected HTTP body: $body"
+                );
+            }
+
+            return '';
```

Comment:
Given that it otherwise returns the JSON decoded content, maybe returning null would make more sense here.